include/eve/detail/function/simd/common/broadcast_group.hpp

```diff
@@ -0,0 +1,101 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/abi.hpp>
+#include <eve/pattern.hpp>
+
+namespace eve::detail
+{
+  template<simd_value Wide, std::ptrdiff_t Group, std::ptrdiff_t Index, std::ptrdiff_t Size>
+  requires(   (Group >  0) && (Group <= std::min(cardinal_v<Wide>, Size))
+          &&  (Index >= 0) && (Index < cardinal_v<Wide>/Group)
+          )
+  EVE_FORCEINLINE auto broadcast_group_ ( EVE_SUPPORTS(cpu_), Wide w
+                                        , fixed<Group> g, index_t<Index> i, fixed<Size> sz
+                                        )
+  {
+    using that_t        = as_wide_t<Wide,fixed<Size>>;
+    using v_t           = eve::element_type_t<Wide>;
+    constexpr auto card = cardinal_v<Wide>;
+
+    [[maybe_unused]] auto const rep = [&](auto v, auto sz)
+    {
+      // If the beginning of the indexed group is beyond half the cardinal
+      if constexpr(Index*Group >= card/2)
+      {
+        // We recompute the index and broadcast something from the upper slice of w
+        return broadcast_group(v.slice(upper_), g , index<Index-card/(Group*2)>, sz);
+      }
+      else
+      {
+        // Else we broadcast somethign from the lower slice of w
+        return broadcast_group(v.slice(lower_), g , i, sz);
+      }
+    };
+
+    // If the output size is equal to the group size and the cardinal, we just return the input
+    if constexpr(Size == Group && Size == card)
+    {
+      return w;
+    }
+    // If the output is aggregated, we do the replication
+    else if constexpr(eve::has_aggregated_abi_v<that_t> )
+    {
+      auto const r = rep(w, lane<(card >= Size ? Size/2 : Size)>);
+      return eve::combine(r,r);
+    }
```

Comment:
Why rep?

Like OK, here is what I understand:

if Group <= 64 bits - do broadcast
else if identity - return
else if aggregated out: size / 2
else  input.splice(), select half

The last thing you cal rep I believe for some reason.

No?

Reply:
Yeah the name is bad, changing to compute_half